File: Philadelphia Workshop/2 - ListEQ/Java_Scaffold/ListEQ.java
import java.util.List;

/* Import Bridges and relevant data source */
import bridges.connect.Bridges;
import bridges.base.SLelement;
import bridges.data_src_dependent.EarthquakeUSGS;

public class ListEQ {

	/* Given an SLelement with an EarthquakeUSGS value, set various visual properties */
	public static void setProperties(SLelement<EarthquakeUSGS> el) {
	    /*TODO: set the properties on the element based on location, magnitude of the earthquake */
	}

	public static void main(String[] args) throws Exception {

		/* Initialize a Bridges connection with your credentials */
		Bridges bridges = new Bridges(2, "bridges_workshop", "1298385986627");

    /* Set an assignment title */
    bridges.setTitle("ListEQ Example");

		/* Get a List of USGS Earthquake Tweet objects from Bridges */
		List<EarthquakeUSGS> mylist = bridges.getEarthquakeUSGSData(100);

		/* Set up a prev and head element */
		SLelement<EarthquakeUSGS> prev = new SLelement<EarthquakeUSGS>();
		SLelement<EarthquakeUSGS> head = new SLelement<EarthquakeUSGS>();

		/* Read each actor movie pair and set up a new SLelement for each */
		for(int i = 0; i < mylist.size(); i++) {

			/* Create each new SLelement */
			SLelement<EarthquakeUSGS> element = new SLelement<EarthquakeUSGS>(mylist.get(i));

			/* Set the element label equal to Title (print it out for sanity check) */
			element.setLabel(mylist.get(i).getTitle());
			System.out.println(element.getLabel());

			/* Pass the element to a function to set its visual properties */
			setProperties(element);

			/* Add 'next' pointer where appropriate */
			if(i > 0) {
				prev.setNext(element);
			} else {
				/* Set the head pointer */
				head = element;
			}
			/* Update the prev pointer */
			prev = element;
		}

		/* Pass the head of the list to Bridges */
		bridges.setDataStructure(head);

		/* Visualize the list */
		bridges.visualize();
	}
}

File: Philadelphia Workshop/2 - ListEQ/C++_Scaffold/ListEQ.cpp
#include "Bridges.h"
#include "DataSource.h"
#include "Array.h"
#include "SLelement.h"


using namespace bridges;

/*
 * Case Sensitive Implementation of endsWith()
 * It checks if the string 'mainStr' ends with given string 'toMatch'
 *
 * From: http://thispointer.com/c-how-to-check-if-a-string-ends-with-an-another-given-string/
 */
bool endsWith(const std::string &mainStr, const std::string &toMatch)
{
	if(mainStr.size() >= toMatch.size() &&
			mainStr.compare(mainStr.size() - toMatch.size(), toMatch.size(), toMatch) == 0)
			return true;
		else
			return false;
}

void setProperties(SLelement<EarthquakeUSGS>* el) {
  /*TODO: set the properties on the element based on location, magnitude of the earthquake */
}

int main() {

	//create the Bridges object, set credentials
	Bridges::initialize(102, "bridges_workshop", "1298385986627");
	
	Bridges::setTitle("List EQ");

	//Get Earthquake data
	std::vector< EarthquakeUSGS > eqs = DataSource::getEarthquakeUSGSData(100);

	//Building linked list
	SLelement<EarthquakeUSGS>* head = nullptr;

	for (auto eq : eqs) {
	  SLelement<EarthquakeUSGS>* newone = new SLelement<EarthquakeUSGS> (eq,
									     eq.getTitle());
	  newone->setNext(head);
	  head = newone;
	}

	//Setting properties.
	{
	  SLelement<EarthquakeUSGS>* current = head;
	  while (current != nullptr) {
	    setProperties(current);
	    current = current->getNext();
	  }
	}

	// tell Bridges what data structure to visualize
	Bridges::setDataStructure(head);

	// visualize the list
	Bridges::visualize();

	//free memory
	while (head != nullptr) {
	  auto next = head->getNext();
	  delete head;
	  head = next;
	}
	
	return 0;
}
